From d24ff856a868bf4f571a47ad4fd5a1b709e75628 Mon Sep 17 00:00:00 2001
From: Akshay Pradip Bharad <apbharad@habana.ai>
Date: Wed, 04 Jun 2025 05:59:39 +0000
Subject: [PATCH] Revert "[SW-227647] ipm: remove typedef"

This reverts commit cdc70aad5dffa4ab736f15c5a2c5dc7ea3aeb300.

Change-Id: Ic67b86d4017884b040f7926940807b0f76c3b666
---

diff --git a/drivers/ipm/ipm_hl.c b/drivers/ipm/ipm_hl.c
index 5684d7c..2a52730 100644
--- a/drivers/ipm/ipm_hl.c
+++ b/drivers/ipm/ipm_hl.c
@@ -17,22 +17,24 @@
 #define LOG_MODULE_NAME			ipm_hl
 LOG_MODULE_REGISTER(LOG_MODULE_NAME, CONFIG_IPM_LOG_LEVEL);
 
+typedef volatile struct ipm_hl_msg_hdr ipm_hl_msg_hdr_t;
+
 /*
  * Beware that this function is called from ISR. It processes the message and
  * sends an ACK to the sender.
  */
-static void process_message(const struct device *dev, volatile struct ipm_hl_msg_hdr *msg)
+static void process_message(const struct device *dev, ipm_hl_msg_hdr_t *msg)
 {
 	struct ipm_hl_data *data = DEV_DATA(dev);
 	uint8_t *addr = (uint8_t *)msg;
 
 	LOG_DBG("MSG RECEIVED: id: %d size: %d", msg->id, msg->size);
-	LOG_HEXDUMP_DBG((addr + sizeof(struct ipm_hl_msg_hdr)), msg->size, "receive");
+	LOG_HEXDUMP_DBG((addr + sizeof(ipm_hl_msg_hdr_t)), msg->size, "receive");
 
 	/* Use zero copy */
 	if (data->callback)
 		data->callback(dev, data->user_data, msg->id,
-			(void *)(addr + sizeof(struct ipm_hl_msg_hdr)));
+			(void *)(addr + sizeof(ipm_hl_msg_hdr_t)));
 	else
 		LOG_ERR("Callback not registered for IPM HL dev %p", dev);
 
@@ -49,11 +51,11 @@
 {
 	const struct ipm_hl_config *config = DEV_CFG(dev);
 	struct ipm_hl_data *data = DEV_DATA(dev);
-	volatile struct ipm_hl_msg_hdr *msg = NULL;
+	ipm_hl_msg_hdr_t *msg = NULL;
 	uint32_t i;
 
 	for (i = 0; i < config->rx.num_non_data_chunks; i++) {
-		msg = (volatile struct ipm_hl_msg_hdr *) (data->rx_ndata_base +
+		msg = (ipm_hl_msg_hdr_t *) (data->rx_ndata_base +
 			   (i * data->chunk_hdr_sz));
 
 		if ((msg->busy) && (!msg->ack)) {
@@ -68,7 +70,7 @@
 	}
 
 	for (i = 0; i < data->num_rx_data_chunks; i++) {
-		msg = (volatile struct ipm_hl_msg_hdr *) (data->rx_data_base +
+		msg = (ipm_hl_msg_hdr_t *) (data->rx_data_base +
 						(i * config->rx.data_chunk_size));
 
 		if ((msg->busy) && (!msg->ack)) {
@@ -118,7 +120,7 @@
 {
 	const struct ipm_hl_config *config = DEV_CFG(dev);
 	struct ipm_hl_data *dev_data = DEV_DATA(dev);
-	volatile struct ipm_hl_msg_hdr *msg = NULL;
+	ipm_hl_msg_hdr_t *msg = NULL;
 	volatile uint8_t *addr, num_chunks;
 	uint32_t i, base, chunk_sz;
 
@@ -146,7 +148,7 @@
 	}
 
 	for (i = 0; i < num_chunks; i++) {
-		msg = (volatile struct ipm_hl_msg_hdr *) (base + (i * chunk_sz));
+		msg = (ipm_hl_msg_hdr_t *) (base + (i * chunk_sz));
 		if ((msg->busy) && (msg->id == id)) {
 			if (!msg->ack) {
 				sys_write32(config->tx.irq_num, (mem_addr_t) config->intr_base);
@@ -160,7 +162,7 @@
 	}
 
 	for (i = 0; i < num_chunks; i++) {
-		msg = (volatile struct ipm_hl_msg_hdr *) (base + (i * chunk_sz));
+		msg = (ipm_hl_msg_hdr_t *) (base + (i * chunk_sz));
 		if (!msg->busy) { /* Chunk is free */
 			break;
 		}
@@ -180,7 +182,7 @@
 	LOG_DBG_LOCAL("MSG CREATED: id: %d size: %d seq no: %d\n",
 			msg->id, msg->size, msg->seq_no);
 
-	memcpy((void *)(addr + sizeof(struct ipm_hl_msg_hdr)), data, size);
+	memcpy((void *)(addr + sizeof(ipm_hl_msg_hdr_t)), data, size);
 
 	HL_ARC_W_BARRIER();
 	/* Raise interrupt */
@@ -234,13 +236,13 @@
 {
 	const struct ipm_hl_config *config = DEV_CFG(dev);
 	struct ipm_hl_data *data = DEV_DATA(dev);
-	volatile struct ipm_hl_msg_hdr *msg = NULL;
+	ipm_hl_msg_hdr_t *msg = NULL;
 	uint32_t i;
 
 	/* TODO: Based on usage, check arch_irq_lock/mutex required for ipm_hl_clear API */
 
 	for (i = 0; i < config->rx.num_non_data_chunks; i++) {
-		msg = (volatile struct ipm_hl_msg_hdr *)
+		msg = (ipm_hl_msg_hdr_t *)
 			(data->rx_ndata_base + (i * data->chunk_hdr_sz));
 		if (msg->id == id) {
 			/* Message is retrieved, Clear id for for TX to send next message */
@@ -253,7 +255,7 @@
 	}
 
 	for (i = 0; i < data->num_rx_data_chunks; i++) {
-		msg = (volatile struct ipm_hl_msg_hdr *) (data->rx_data_base +
+		msg = (ipm_hl_msg_hdr_t *) (data->rx_data_base +
 						(i * config->rx.data_chunk_size));
 		if (msg->id == id) {
 			/* Message is retrieved, Clear id for for TX to send next message */
@@ -279,7 +281,7 @@
 	if (config->irq_config_func)
 		config->irq_config_func(dev);
 
-	data->chunk_hdr_sz = sizeof(volatile struct ipm_hl_msg_hdr); /* 12 bytes */
+	data->chunk_hdr_sz = sizeof(ipm_hl_msg_hdr_t); /* 12 bytes */
 	if (config->mbx_type == MBX_TX_ONLY || config->mbx_type == MBX_BIDIRECTIONAL) {
 		data->tx_non_data_base = config->tx.mbx_base;
 		data->tx_data_base = data->tx_non_data_base +
